### PR TITLE
Online user manual fixes

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: doc/cprover-manual/properties.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+specify program properties. Assertions are properties of the state of
+the program when the program reaches a particular program location.
+Assertions are often written by the programmer by means of the `assert`
+macro.
+
+In addition to the assertions written by the programmer, assertions for
+specific properties can also be generated automatically by CBMC, often
+relieving the programmer from writing "obvious" assertions.
+
+CBMC come with an assertion generator, which performs a conservative [static
```
Comment (tautschnig) [inline doc/cprover-manual/properties.md]: comes

File: doc/cprover-manual/properties.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+CBMC uses
+[assertions](http://en.wikipedia.org/wiki/Assertion_(computing)) to
+specify program properties. Assertions are properties of the state of
+the program when the program reaches a particular program location.
+Assertions are often written by the programmer by means of the `assert`
+macro.
+
+In addition to the assertions written by the programmer, assertions for
+specific properties can also be generated automatically by CBMC, often
+relieving the programmer from writing "obvious" assertions.
```
Comment (tautschnig) [inline doc/cprover-manual/properties.md]: often relieving the programmer from expressing properties that should hold in any well-behaved program.

File: doc/cprover-manual/properties.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+the following properties:
+
+-   **Buffer overflows.** For each array access, check whether the upper
+    and lower bounds are violated.
+-   **Pointer safety.** Search for `NULL`-pointer dereferences or
+    dereferences of other invalid pointers.
+
+-   **Division by zero.** Check whether there is a division by zero in
+    the program.
+
```
Comment (tautschnig) [inline doc/cprover-manual/properties.md:42]: Missing: signed/unsigned overflow, memory leaks

File: doc/cprover-manual/properties.md [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+-   **Not-a-Number.** Check whether floating-point computation may
+    result in [NaNs](http://en.wikipedia.org/wiki/NaN).
+
+-   **Unitialized local.** Check whether the program uses an
+    uninitialized local variable.
+
+-   **Data race.** Check whether a concurrent program accesses a shared
+    variable at the same time in two threads.
```
Comment (tautschnig) [inline doc/cprover-manual/properties.md]: I'm not sure whether we should mention those last two in here, they aren't particularly well maintained or tested.

File: doc/cprover-manual/properties.md [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+```
+
+We can now get a list of the assertions that have been generated as
+follows:
+
+```
+    goto-instrument out.gb --show-properties
+```
+
+Using either CBMC or SATABS on `out.gb`, we can obtain a counterexample
```
Comment (tautschnig) [inline doc/cprover-manual/properties.md]: Remove SATABS?

File: doc/cprover-manual/properties.md [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+
+Flag                         |  Check
+-----------------------------|----------------------------------------------
+`--no-assertions`            |  ignore user assertions
+`--bounds-check`             |  add array bounds checks
+`--div-by-zero-check`        |  add division by zero checks
+`--pointer-check`            |  add pointer checks
+`--signed-overflow-check`    |  add arithmetic over- and underflow checks
+`--unsigned-overflow-check`  |  add arithmetic over- and underflow checks
+`--undefined-shift-check`    |  add range checks for shift distances
```
Comment (tautschnig) [inline doc/cprover-manual/properties.md:128]: Ah, also missing in the above list

File: doc/cprover-manual/properties.md [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+    goto-instrument error_example.goto error_example_replaced.goto \
+      --generate-function-body '(?!__).*_error' \
+      --generate-function-body-options assert-false
+    cbmc error_example_replaced.goto
+
+Which gets us the output
+
+> ** Results:
+> [internal_error.assertion.1] assertion false: FAILURE
+> [api_error.assertion.1] assertion false: FAILURE
```
Comment (tautschnig) [inline doc/cprover-manual/properties.md]: Line numbers

**Review by tautschnig — APPROVED** [review]

File: doc/cprover-manual/installation.md [hunk 8]
```diff
@@ -27,10 +27,11 @@ and the directories that contain the header files. You must run CBMC
 from within the *Visual Studio Command Prompt*.
 
 Note that the distribution files for the [Eclipse
-plugin](installation-plugin.shtml) include the CBMC executable.
-Therefore, if you intend to run CBMC exclusively within Eclipse, you can
-skip the installation of the CBMC executable. However, you still have to
-install the compiler environment as described above.
+plugin](https://raw.githubusercontent.com/diffblue/cbmc/develop/doc/cprover-manual/index.md)
```
Comment (johnnonweiler) [inline doc/cprover-manual/installation.md]: That URL doesn't point to the Eclipse Plugin.

**Comment by peterschrammel** [comment]
@kroening, does this already auto-synchronise with http://www.cprover.org/cprover-manual/ ?

**Comment by johnnonweiler** [comment]
@peterschrammel http://www.cprover.org/cprover-manual/ shows an old version.  (Compare the index there to this one on cbmc/develop https://github.com/diffblue/cbmc/blob/develop/doc/cprover-manual/index.md .)

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 37c2a68017979984ea753f78baf5e81cbd93eb82).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91779159

**Comment by kroening** [comment]
@peterschrammel Yes, this is now active. We do need to be somewhat conscious to keep URLs stable.

**Pull request merged**